sui_programmability/examples/basics/sources/Sandwich.move

```diff
@@ -62,8 +63,9 @@ module Basics::Sandwich {
         c: Coin<SUI>,
```

Comment:
One thing we probably want to do is to make (or add) the `withdraw` method in Coin to spill out Balance.
This is because I expect in most cases, entry functions like this would not take an entire coin by value (because that requires them to split it first), and more likely they want to pass coin by mut ref, and in that case you would want to be able to withdraw.

Reply:
Re passing a mutable reference to a Coin - I don't think I'd ever authorize a transaction which can withdraw any amount from my mutable coin. It doesn't seem secure, and if there's a bug or an intention to steal my funds, they can withdraw the whole amount.

Reply:
Good point. So I guess most tx that involve coins will have to go through two transactions

Reply:
Yeah, that sounds like it. If only we had lambdas, so that we can withdraw before we call a generic callback. 😆